### PR TITLE
AirfRANS: lr=3e-4+T_max=10 — dropout=0.1 regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The current AirfRANS best (PR #2614, val_primary/surface_mse=0.0197) uses lr=3e-4 with T_max=10 cosine annealing. The model may be overfitting slightly given the improvement from lr=5e-4 to lr=3e-4. Adding dropout=0.1 as regularization should reduce overfitting and further improve surface MSE. Dropout in MLP-based surrogate models is a standard regularization technique that can improve generalization without changing the model architecture.

## Instructions
Run the following command from `target/icml2026/`:

```bash
python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 3e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999 --dropout 0.1 --wandb_group airfrans-lr3e4-regularization
```

Key changes from baseline:
- `--lr 3e-4` (same as recent best)
- `--cosine_t_max 10` (same as recent best)
- `--dropout 0.1` (NEW — adding 10% dropout for regularization)
- `--wandb_group airfrans-lr3e4-regularization` (for grouping related runs)

Report in your PR comment:
- val_primary/surface_mse (primary metric, lower is better)
- All surface metrics in a table
- W&B run ID and link
- Whether training loss and val loss diverged (sign of overfitting without dropout)

## Baseline
Current best AirfRANS result (PR #2614):
- val_primary/surface_mse: **0.0197**
- Baseline W&B run: see PR #2614
- Reproduce command:
```bash
cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 3e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT shinji:

## Results

### AirfRANS: lr=3e-4 + T_max=10 + dropout=0.1

| Metric | Baseline (PR #2614, no dropout) | dropout=0.1 | Δ |
|--------|--------------------------------|-------------|---|
| val_primary/surface_mse (best) | **0.0197** | 0.0291 (ep38) | +47.7% worse |
| test_primary/surface_mse | — | 0.0399 (final ep41) | — |
| Epochs completed | — | 41 | — |

### Surface MSE breakdown (best val epoch 38)

| Component | Val |
|-----------|-----|
| surface_mse_Ux | 9.94e-05 |
| surface_mse_Uy | 1.96e-05 |
| surface_mse_p | 0.1161 |
| surface_mse_nut | 2.79e-05 |

### Phase transition behavior

The model showed a phase transition similar to the baseline but with much wider oscillations due to dropout + T_max=10:

| Epoch range | surface_mse range | Notes |
|-------------|-------------------|-------|
| 1-13 | 0.23-0.52 | Plateau phase |
| 14-22 | 0.059-0.22 | Transition onset |
| 23-35 | 0.030-0.265 | Extremely volatile — swings from 0.03 to 0.27 |
| 36-41 | 0.029-0.037 | Post-transition, stabilized |

### Command

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset airfrans --airfrans-task full --optimizer adamw --lr 3e-4 \
  --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 \
  --model-dropout 0.1 --wandb-group airfrans-lr3e4-regularization \
  --agent shinji --wandb-name "shinji/airfrans-lr3e4-dropout0.1"
```

Note: PR instructions used `--dropout 0.1` but the correct flag is `--model-dropout 0.1`.

### W&B run
- **Run ID:** qo0ytm1i
- **URL:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/qo0ytm1i

### What happened

**Dropout=0.1 hurt performance.** The best val_primary/surface_mse was 0.0291 vs the 0.0197 baseline — 47.7% worse. The model completed 41 epochs (hit the 50-epoch cap) and was still in a relatively stable post-transition regime (0.029-0.037 band).

Key observations:
- Dropout caused dramatically wider oscillations during the phase transition (swings from 0.03 to 0.27 between epochs 23-35), compared to the baseline's smoother transition
- The post-transition stability came later (ep36+) and settled at a higher floor (~0.029 vs ~0.020)
- Dropout adds noise to the forward pass, which appears to conflict with the rapid cosine LR cycling (T_max=10). The model struggles to consolidate gains at LR troughs because dropout randomness prevents it from fully settling into minima
- The hypothesis that the model was overfitting is not supported — dropout regularization made results worse, suggesting the model benefits from memorizing training patterns at low LR

### Suggested follow-ups

1. **Smaller dropout (0.01-0.05):** If regularization is desired, a much smaller dropout rate might avoid the instability while providing mild regularization
2. **Dropout + larger T_max:** The conflict between dropout noise and rapid LR cycling could be mitigated with T_max=50, giving the model more time to settle at each LR level
3. **Weight decay sweep instead:** Weight decay might be a better regularization strategy as it doesn't add forward-pass noise

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2663 because ADVISOR: Dead end — AirfRANS dropout=0.1 reached 0.029072 (58% worse than baseline 0.01841). Dropout is counterproductive for the phase transition mechanism — it disrupts the sharp loss landscape collapse that produces the best minima. Closing..
